### PR TITLE
New feature: `split`

### PR DESCRIPTION
**Pull request by noahlaux**
Introduces `split` to built in string functions enabling string splitting with a delimeter. Input and delimeter can either be `string` or `number`.

Discussed in #368 

# Checklist

Before submitting a pull request, please check the following:

* [x] All tests pass for you on your machine for all implementations (all implementations must behave identically)
* [x] New tests have been added for any new functionality or to prevent regressions of a bugfix
* [x] Added a short description of the change to `newsfragments/$issue.bugfix` (for fixes) or `.feature` (for new features) or `.doc` (for documentation-only changes)

### CODE REVIEW

File: specification.yml [hunk 1]
```diff
@@ -1940,6 +1940,41 @@ context: {}
...
+---
+title: split string with no match
+context: {}
+template: {$eval: "split('no semicolon', ':')"}
+result: ['no semicolon']
+---
+title: split with number delimiter
+context: {}
+template: {$eval: "split('left1right', 1)"}
+result: ['left', 'right']
```
Comment (djmitche) [inline specification.yml:1966]: Is there a rationale for this particular behavior?
```
>>> 'abc123'.split(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: must be str or None, not int
```

Comment (noahlaux) [reply specification.yml:1966]: My rationale was that the end user didn't particularly want to think about casting in the context of json-e, and if I don't think there's a huge risk of non deterministic behavior. IMHO, it doesn't matter if the user wanna split on `"1"` or `1`, more the intent.

`split('left1right', 1)` vs `split('left1right', str(1))`

I think the latter just adds noise or I'm not too concerned about failure, though I can be mistaken.

Comment (djmitche) [reply specification.yml:1966]: What do you think, @helfi92?

Comment (helfi92) [reply specification.yml:1966]: I agree. We can move forward with the current implementation. Worst case we just revisit this in the future if an issue is raised.

**Pull request merged**